verification_token/models.py

```diff
@@ -19,14 +19,35 @@ class VerificationTokenManager(models.Manager):
     def deactivate(self, obj, slug=None, key=None):
         self.filter_active_tokens(obj, slug, key).update(is_active=False)
 
-    def deactivate_and_create(self, obj, slug=None, extra_data=None, deactivate_old_tokens=True, expiration_in_minutes=None,
-                              key_generator_kwargs=None):
-        expiration_in_minutes = settings.DEFAULT_EXPIRATION if expiration_in_minutes is None else expiration_in_minutes
+    def deactivate_and_create(self, obj, slug=None, extra_data=None, deactivate_old_tokens=True,
+                              key_generator_kwargs=None, **kwargs):
+        expiration_in_minutes = (settings.DEFAULT_EXPIRATION if 'expiration_in_minutes' not in kwargs.keys()
+                                 else kwargs['expiration_in_minutes']
+        )
 
-        key_generator_kwargs = {} if key_generator_kwargs is None else key_generator_kwargs
         if deactivate_old_tokens:
             self.deactivate(obj, slug)
 
+        return self._create(obj, slug=slug, extra_data=extra_data, key_generator_kwargs=key_generator_kwargs,
+                            expiration_in_minutes=expiration_in_minutes
+        )
+
+    def get_or_create(self, obj, slug=None, extra_data=None, key=None, key_generator_kwargs=None, **kwargs):
```

Comment:
radši raději bych defaultní nepřetěžoval, aby nedošlo k nějakému zmatku. Pojmenoval bych get_active_or_create

Reply:
OK, to dává smysl.

Reply:
Upraveno.